server/src/main/java/org/opensearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -257,9 +272,77 @@ public DocValueFormat docValueFormat(@Nullable String format, ZoneId timeZone) {
             return DocValueFormat.BOOLEAN;
         }
 
+        @Override
+        public Query termQuery(Object value, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            Query query = new TermQuery(new Term(name(), indexedValueForSearch(value)));
+            if (boost() != 1f) {
```

Comment:
Why do we need to treat boosting here in any special way?

Reply:
All queries can have boosts applied to them. I believe if we have a different boost than the default, we would need to wrap the query in a boost query to ensure that the boosting value applies.

Reply:
Oh I see, it was missed indeed, could we please add the test for it?

Reply:
Added a new case in the unit tests for `BooleanFieldMapper`